lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +73,130 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+
+        with open(self.path_to_config, 'r', encoding='utf-8') as json_file:
+            self.content = json.load(json_file)
+
+        self._seed_urls = self.content['seed_urls']
+        self._num_articles = self.content['total_articles_to_find_and_parse']
+        self._headers = self.content['headers']
+        self._encoding = self.content['encoding']
+        self._timeout = self.content['timeout']
+        self._should_verify_certificate = self.content['should_verify_certificate']
+        self._headless_mode = self.content['headless_mode']
+
+        self._validate_config_content()
+        self.config_obj = self._extract_config_content()
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+
+        return ConfigDTO(self._seed_urls,
+                         self._num_articles,
+                         self._headers,
+                         self._encoding,
+                         self._timeout,
+                         self._should_verify_certificate,
+                         self._headless_mode)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        if not isinstance(self._seed_urls, list):
+            raise IncorrectSeedURLError
+
+        for element in self._seed_urls:
+            if not isinstance(element, str):
+                raise IncorrectSeedURLError
+
+            if 'https://' not in element:
+            #if not re.match(r'https://', element):
+                raise IncorrectSeedURLError
+
+        if not isinstance(self._num_articles, int)\
+                or isinstance(self._num_articles, bool) or self._num_articles <= 0:
```

Comment:
why do you check for bool??

Reply:
because if you pass a `True` value to `isinstance(value, int)`, it will return True, and i don't need the number of articles to be `True` instead of an integer